ApollosProject/apollos-prototype
# Clean up / polish navigation on the connect tab

## DESCRIPTION

### What does this PR do, or why is it needed?

- Cleans up navigation on user settings forms by adding a navigation header and tweaking where in the stack the navigation happens
- Tweaks/simplifies the component structure on the personal details + change password form. Primary output is that this decreases code debt
- Fixes typo on "Recently Liked" header on connect tab
- Adds navigation header to Recently Liked feed
- Fixes data loading on recently liked feed. Previously, not all information was showing up (the query was incomplete)

### What design trade-offs/decisions were made?

The main trade-offs made had to do with the design on the forms. Mainly with the way the cancel and Save buttons worked. Currently on master, these forms have a single "Cancel" or "Done" button in the upper right hand. However, it's not clear what happens when you swipe down or if you want to cancel your changes after you've made some. We need to have two clear actions: a cancel action, and a save/complete/done action. 

In order to accomplish this with the least code debt, I opted to add a `Save` button at the bottom of the screen,  and a `Cancel` button inside of the navigation header. Our `<Button/>` component cleanly handles the different loading/validation states of our forms, so there is a usability win in using `<Button />`. I tried to add a "Save" text button at the top right hand corner inside of the navigation header, but there's not a clean way to do that with `react-navigation` (due to component scope). 

I also tried several different variations of using no navigation header, making this component a modal with a close button in the top right, etc. I believe this ended up being the best solution.

![image](https://user-images.githubusercontent.com/103927/47463722-a5c07380-d7ac-11e8-8d7c-7efa14bd518f.png)

### How do I test this PR?

Sign in, test out the forms. Make sure the software keyboard is visible. Make sure layout still looks okay on smaller devices, especially android.

### What automated tests did you write?

Refactored existing components. Existing components don't have tests, but there were snapshot changes.

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
